weblate/middleware.py

```diff
@@ -180,6 +184,21 @@ def process_exception(self, request, exception):
                     return None
                 kwargs["component"] = component.slug
 
+            if kwargs.get("new_language"):
+                new_lang = kwargs.get("new_language")
+                if new_lang not in project.languages:
+                    messages.add_message(
+                        request,
+                        messages.INFO,
+                        _(f"{new_lang} is currently not available, but can be added."),
+                    )
+                    return redirect(
+                        reverse(
+                            "component", args=[kwargs["project"], kwargs["component"]]
```

Comment:
The code is reachable when component is not set....

Reply:
Thanks for the catch. Updated.

I am not sure how weblate shows info messages to the User. I used messages for this. let me know if we have something already.

Reply:
Using messages is the correct way to go.